### PR TITLE
Feature Update: Implementation of Person, Student, and Teacher Classes

### PR DESCRIPTION
**Pull request by cancelei**
**Highlights:**

**Person Class:**

- Introduced instance variables: `@id`, `@name`, and `@age`.
- Constructor accepts name, age, and parent_permission. Default values: name = "Unknown", parent_permission = true.
- Provided getters for `@id`, `@name`, and `@age`; and setters for `@name` and `@age`.
- Introduced a private method of_age? determining age threshold.
- Public method can_use_services? checks age or parental permission for service usage.

**Student Class:**

- Inherits from the Person class.
- The constructor is extended to include @classroom with a corresponding parameter.
- New method play_hooky implemented.

**Teacher Class:**

- Inherits from the Person class.
- The constructor is extended to include `@specialization` with a corresponding parameter.
- Overrode the can_use_services? method to always return true.

**File Organization:**

- Each class has been saved in a separate file to maintain clarity and modularity.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
+# Person Class
 class Person
   attr_reader :id
   attr_accessor :name, :age
 
-  def initialize(id, name = "Unknown", age = 0)
+  def initialize(id, name = 'Unknown', age = 0)
```
Comment (Efiamotu-1) [inline person.rb]: - [x] I want to kindly suggest that for your constructor function, you do not need users to pass in an `id` rather this should automatically be generated when a new person is created, see the example code below for how to achieve this
- [x] Also one more parameter you should requires to create a new person should be the `parent_permission` which by default should be equal to `true`
- [x] Also the age parameter does  not require a default value
- >* example code 
```ruby
def initialize(age, name = "Unknown", parent_permission = true)
    @id = Random.rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
 end
```

File: student.rb [hunk 2]
```diff
@@ -1,14 +1,15 @@
+# Student Class
 require_relative 'person'
 
 class Student < Person
   attr_accessor :classroom
 
-  def initialize(id, classroom, name = "Unknown", age = 0)
+  def initialize(id, classroom, name = 'Unknown', age = 0)
```
Comment (Efiamotu-1) [inline student.rb]: - [x] kindly update this as well to not pass in the `id` but pass in the `parent_permission` parameter 
- >* example code 
```ruby
def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
```

File: teacher.rb [hunk 3]
```diff
@@ -1,14 +1,15 @@
+# Teacher Class
 require_relative 'person'
 
 class Teacher < Person
   attr_accessor :specialization
 
-  def initialize(id, specialization, name = "Unknown", age = 0)
+  def initialize(id, specialization, name = 'Unknown', age = 0)
```
Comment (Efiamotu-1) [inline teacher.rb]: - [x] Kindly add the same update to the parameters as done in the student and person class 👍

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @cancelei , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @cancelei , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done implementing the requested changes. 🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**